test/e2e/storage/testsuites/base.go

```diff
@@ -64,26 +64,38 @@ func getTestNameStr(suite TestSuite, pattern testpatterns.TestPattern) string {
 	return fmt.Sprintf("[Testpattern: %s]%s %s%s", pattern.Name, pattern.FeatureTag, tsInfo.name, tsInfo.featureTag)
 }
 
-// RunTestSuite runs all testpatterns of all testSuites for a driver
+// RunTestSuite defines tests for all testpatterns and all testSuites for a driver
 func RunTestSuite(f *framework.Framework, driver TestDriver, tsInits []func() TestSuite, tunePatternFunc func([]testpatterns.TestPattern) []testpatterns.TestPattern) {
 	for _, testSuiteInit := range tsInits {
 		suite := testSuiteInit()
 		patterns := tunePatternFunc(suite.getTestSuiteInfo().testPatterns)
 
 		for _, pattern := range patterns {
+			if !isTestSupported(suite, driver, pattern) {
+				continue
+			}
+
+			if bTestDriver, ok := driver.(BeforeEachTestDriver); ok {
+				p := pattern
+				BeforeEach(func() {
+					bTestDriver.BeforeEach(p)
+				})
+			}
+
 			suite.execTest(driver, pattern)
 		}
 	}
 }
 
-// skipUnsupportedTest will skip tests if the combination of driver, testsuite, and testpattern
-// is not suitable to be tested.
+// isTestSupported will determine if the combination of driver, testsuite, and testpattern
+// can be tested.
+//
 // Whether it needs to be skipped is checked by following steps:
```

Comment:
"Whether *the test combination is supported* is checked..."

Reply:
Done.